game_engine/replay/GameEvent.cpp

```diff
@@ -60,8 +68,15 @@ void CollisionEvent::update_stats(const Store &store, const Map &map,
     ordered_id_map<Player, int> ships_involved;
     for (const auto &ship_id : ships) {
         const auto &entity = store.get_entity(ship_id);
-        stats.player_statistics.at(entity.owner.value).all_collisions++;
+        auto &player_stats = stats.player_statistics.at(entity.owner.value);
+        
         ships_involved[entity.owner]++;
+        player_stats.all_collisions++;
+        player_stats.total_dropped += entity.energy;
+        
+        if (map.at(location).owner != Player::None) { // There is a dropoff
+            player_stats.dropoff_collisions++;
+        }
```

Comment:
Should `dropoff_collisions` really count collisions over all dropoffs? It seems like only counting collisions over the player's own dropoffs might be more useful.